model/config.go

```diff
@@ -161,6 +161,32 @@ func (i *IstioRegistry) RouteRules(namespace string) []*proxyconfig.RouteRule {
 	return out
 }
 
+// RouteRulesBySource selects routing rules from source to destination (sorted by precedence)
+func (i *IstioRegistry) RouteRulesBySource(namespace string, instances []*ServiceInstance) []*proxyconfig.RouteRule {
+	out := make([]*proxyconfig.RouteRule, 0)
+	for _, rule := range i.RouteRules(namespace) {
+		if rule.Match != nil {
```

Comment:
(optional) you could invert the `rule.match != nil` check here and get rid of the extra `found` variable, e.g.

```golang
if rule.Match == nil {
    continue
}
for _, instance := range instances {
    if rule.Match.Source != "" && rule.Match.Source != instance.Service.Hostname {
        continue
    }
    var tags Tags = rule.Match.SourceTags
    if !tags.SubsetOf(instance.Tags) {
        continue
    }
    out = append(out, rule)
}
```